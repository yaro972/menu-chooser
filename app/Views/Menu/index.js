import MenuView from "./MenuView";

export const Menu = MenuView;
