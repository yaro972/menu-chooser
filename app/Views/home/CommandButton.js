import styles from "./home.style";
import {Text, TouchableOpacity} from "react-native";
import React from "react";

export const CommandButton = ({onPress}) => {
    return (
        <TouchableOpacity
            onPress={onPress}
            style={styles.buttonCommand}
            activeOpacity={0.7}
        >
            <Text style={styles.textButtonCommand}>Commander</Text>
        </TouchableOpacity>
    )
};
