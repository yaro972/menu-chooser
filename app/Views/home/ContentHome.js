import React from 'react';
import {Text, View} from "react-native";
import styles from "./home.style";

export const ContentHome = (props) => {
    return (
        <View style={styles.containerTitre}>
            <Text style={[styles.mainTextTitleWhite, styles.h1Titre]}>Au nem
                d'argent</Text>
            <Text style={[styles.mainTextTitleWhite, styles.address]}>2 rue
                d'along 57070 Metz, France</Text>
            <Text style={[styles.mainTextTitleWhite, styles.address]}>06 28 65
                56 65</Text>
            <View
                style={styles.separator}
            />
            <Text style={[
                styles.mainTextTitleWhite,
                styles.openTitle]}>HORAIRES</Text>
            <Text style={[styles.mainTextTitleWhite]}>Ouvert 7/7 de 11h30 Ã 
                23h30</Text>

            <View style={styles.buttonCommandContainer}>

                {props.children}
            </View>
        </View>
    )
};
