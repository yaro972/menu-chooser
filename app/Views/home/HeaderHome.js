import {Image, View} from "react-native";
import styles from "./home.style";
import React from "react";

export const HeaderHome = () => {
    return (
        <View style={styles.containerImage}>
            <Image
                source={require(
                    '../../assets/img/vietnam_restaurant_300.jpg')}
                style={styles.mainImage}/>
        </View>

    )
};
