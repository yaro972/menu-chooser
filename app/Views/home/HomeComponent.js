import React from 'react';
import {
  Image,
  Text,
  TouchableOpacity,
  View,
} from 'react-native';

import styles from './home.style';

class HomeComponent extends React.Component {
  render() {
    return (

        <View style={styles.container}>
          <View style={styles.containerImage}>
            <Image
                source={require(
                    '../../assets/img/vietnam_restaurant_300.jpg')}
                style={styles.mainImage}/>
          </View>

          <View style={styles.containerTitre}>
            <Text style={[styles.mainTextTitleWhite, styles.h1Titre]}>Au nem
              d'argent</Text>
            <Text style={[styles.mainTextTitleWhite, styles.address]}>2 rue
              d'along 57070 Metz, France</Text>
            <Text style={[styles.mainTextTitleWhite, styles.address]}>06 28 65
              56 65</Text>
            <View
                style={styles.separator}
            />
            <Text style={[
              styles.mainTextTitleWhite,
              styles.openTitle]}>HORAIRES</Text>
            <Text style={[styles.mainTextTitleWhite]}>Ouvert 7/7 de 11h30 Ã 
              23h30</Text>

            <View style={styles.buttonCommandContainer}>

              <TouchableOpacity
                  onPress={() => this.props.navigation.navigate('MenuRoute')}
                  style={styles.buttonCommand}
                  activeOpacity={0.7}
              >
                <Text style={styles.textButtonCommand}>Commander</Text>
              </TouchableOpacity>
            </View>
          </View>
        </View>
    );
  }
}

export default HomeComponent;
