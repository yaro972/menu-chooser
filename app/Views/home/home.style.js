import {StyleSheet} from 'react-native';

const styles = StyleSheet.create(
    {
        container: {
            flex: 1,
        },
        containerImage: {
            flex: 1,
            backgroundColor: '#FF0000',
        },
        containerTitre: {
            flex: 1,
            backgroundColor: '#1a1a1a',
            paddingTop: 10,
            padding: 10,
        },
        mainImage: {
            height: '100%',
            width: '100%',
        },
        mainTextTitleWhite: {
            color: '#c0d1ce',
        },
        h1Titre: {
            fontSize: 30,
            textTransform: 'uppercase',
        },
        address: {
            fontSize: 14,
        },
        separator: {
            borderBottomColor: 'white',
            borderBottomWidth: 1,
            marginTop: 20,
            width: 80,
        },
        openTitle: {
            marginTop: 5,
            fontSize: 18,
        },
        buttonCommandContainer: {
            marginTop: 50,
            alignItems: 'center',
        },
        buttonCommand: {
            backgroundColor: '#c61e1e',
            paddingTop: 8,
            paddingBottom: 10,
            paddingLeft: 40,
            paddingRight: 40,
            textAlign: 'center',
            borderRadius: 7,
        },
        textButtonCommand: {
            color: '#eaf05e',
            fontSize: 18,
            textTransform: 'uppercase',
        },
    }
);

export default styles;
