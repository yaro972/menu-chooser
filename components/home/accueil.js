import React from 'react';
import {
  View,
  Text,
  Image,
} from 'react-native';
import styles from './home.style';

export default class HomeComponent extends React.Component {
  render() {
    return (
        <View Style={styles.container}>
          <Text>Hello</Text>
        </View>
    );
  }
}
