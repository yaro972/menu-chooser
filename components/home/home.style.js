import {Stylesheet} from 'React-native';

const styles = Stylesheet.create({
    container : {
        flex : 1
    }
});

export default styles;