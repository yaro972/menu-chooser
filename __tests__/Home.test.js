// __tests__/Intro-test.js
import React from 'react';
import { shallow } from 'enzyme';
import renderer from 'react-test-renderer';
import {Home} from "../app/Views/home";
import {CommandButton} from "../app/Views/home/CommandButton";

describe('Home Component', () => {
    test('renders correctly', () => {
        const tree = renderer.create(<Home/>).toJSON();
        expect(tree).toMatchSnapshot();
    });
});


describe('Command button', () => {
    const clickCallBack = jest.fn();

    const btn = shallow(<CommandButton onPress={() => clickCallBack()}/>);
    btn.find('')

    console.log('eeee', btn);
        // TODO VÃ©rification appel click
});
