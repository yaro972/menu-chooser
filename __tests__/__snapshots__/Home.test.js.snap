// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Home Component renders correctly 1`] = `
<View
  style={
    Object {
      "flex": 1,
    }
  }
>
  <View
    style={
      Object {
        "backgroundColor": "#FF0000",
        "flex": 1,
      }
    }
  >
    <Image
      source={
        Object {
          "testUri": "../../../app/assets/img/vietnam_restaurant_300.jpg",
        }
      }
      style={
        Object {
          "height": "100%",
          "width": "100%",
        }
      }
    />
  </View>
  <View
    style={
      Object {
        "backgroundColor": "#1a1a1a",
        "flex": 1,
        "padding": 10,
        "paddingTop": 10,
      }
    }
  >
    <Text
      style={
        Array [
          Object {
            "color": "#c0d1ce",
          },
          Object {
            "fontSize": 30,
            "textTransform": "uppercase",
          },
        ]
      }
    >
      Au nem d'argent
    </Text>
    <Text
      style={
        Array [
          Object {
            "color": "#c0d1ce",
          },
          Object {
            "fontSize": 14,
          },
        ]
      }
    >
      2 rue d'along 57070 Metz, France
    </Text>
    <Text
      style={
        Array [
          Object {
            "color": "#c0d1ce",
          },
          Object {
            "fontSize": 14,
          },
        ]
      }
    >
      06 28 65 56 65
    </Text>
    <View
      style={
        Object {
          "borderBottomColor": "white",
          "borderBottomWidth": 1,
          "marginTop": 20,
          "width": 80,
        }
      }
    />
    <Text
      style={
        Array [
          Object {
            "color": "#c0d1ce",
          },
          Object {
            "fontSize": 18,
            "marginTop": 5,
          },
        ]
      }
    >
      HORAIRES
    </Text>
    <Text
      style={
        Array [
          Object {
            "color": "#c0d1ce",
          },
        ]
      }
    >
      Ouvert 7/7 de 11h30 à 23h30
    </Text>
    <View
      style={
        Object {
          "alignItems": "center",
          "marginTop": 50,
        }
      }
    >
      <View
        accessible={true}
        isTVSelectable={true}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          Object {
            "backgroundColor": "#c61e1e",
            "borderRadius": 7,
            "opacity": 1,
            "paddingBottom": 10,
            "paddingLeft": 40,
            "paddingRight": 40,
            "paddingTop": 8,
            "textAlign": "center",
          }
        }
      >
        <Text
          style={
            Object {
              "color": "#eaf05e",
              "fontSize": 18,
              "textTransform": "uppercase",
            }
          }
        >
          Commander
        </Text>
      </View>
    </View>
  </View>
</View>
`;

exports[`renders correctly 1`] = `
<View
  style={
    Object {
      "flex": 1,
    }
  }
>
  <View
    style={
      Object {
        "backgroundColor": "#FF0000",
        "flex": 1,
      }
    }
  >
    <Image
      source={
        Object {
          "testUri": "../../../app/assets/img/vietnam_restaurant_300.jpg",
        }
      }
      style={
        Object {
          "height": "100%",
          "width": "100%",
        }
      }
    />
  </View>
  <View
    style={
      Object {
        "backgroundColor": "#1a1a1a",
        "flex": 1,
        "padding": 10,
        "paddingTop": 10,
      }
    }
  >
    <Text
      style={
        Array [
          Object {
            "color": "#c0d1ce",
          },
          Object {
            "fontSize": 30,
            "textTransform": "uppercase",
          },
        ]
      }
    >
      Au nem d'argent
    </Text>
    <Text
      style={
        Array [
          Object {
            "color": "#c0d1ce",
          },
          Object {
            "fontSize": 14,
          },
        ]
      }
    >
      2 rue d'along 57070 Metz, France
    </Text>
    <Text
      style={
        Array [
          Object {
            "color": "#c0d1ce",
          },
          Object {
            "fontSize": 14,
          },
        ]
      }
    >
      06 28 65 56 65
    </Text>
    <View
      style={
        Object {
          "borderBottomColor": "white",
          "borderBottomWidth": 1,
          "marginTop": 20,
          "width": 80,
        }
      }
    />
    <Text
      style={
        Array [
          Object {
            "color": "#c0d1ce",
          },
          Object {
            "fontSize": 18,
            "marginTop": 5,
          },
        ]
      }
    >
      HORAIRES
    </Text>
    <Text
      style={
        Array [
          Object {
            "color": "#c0d1ce",
          },
        ]
      }
    >
      Ouvert 7/7 de 11h30 à 23h30
    </Text>
    <View
      style={
        Object {
          "alignItems": "center",
          "marginTop": 50,
        }
      }
    >
      <View
        accessible={true}
        isTVSelectable={true}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          Object {
            "backgroundColor": "#c61e1e",
            "borderRadius": 7,
            "opacity": 1,
            "paddingBottom": 10,
            "paddingLeft": 40,
            "paddingRight": 40,
            "paddingTop": 8,
            "textAlign": "center",
          }
        }
      >
        <Text
          style={
            Object {
              "color": "#eaf05e",
              "fontSize": 18,
              "textTransform": "uppercase",
            }
          }
        >
          Commander
        </Text>
      </View>
    </View>
  </View>
</View>
`;
